src/init.cpp

```diff
@@ -1866,6 +1866,35 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
+#ifdef __APPLE__
+    // Warn about exFAT filesystem usage on macOS
+    std::vector<std::string> exfat_paths;
+    std::vector<std::string> error_paths;
+
+    auto check_fs = [&](const fs::path& path, std::string_view desc) {
+        FSType fs_type = GetFilesystemType(path);
+        std::string path_desc = strprintf("%s (\"%s\")", desc, fs::PathToString(path));
+        if (fs_type == FSType::EXFAT) {
+            exfat_paths.push_back(path_desc);
+        } else if (fs_type == FSType::ERROR) {
+            error_paths.push_back(path_desc);
+        }
+    };
+
+    check_fs(args.GetDataDirNet(), "data directory");
+    check_fs(args.GetBlocksDirPath(), "blocks directory");
+
+    if (!exfat_paths.empty()) {
+        InitWarning(strprintf(_("The following paths are on exFAT which is known to have intermittent corruption problems on macOS: %s. "
```

Comment:
nit: "are on exFAT" sounds weird to me + missing comma:
```suggestion
        InitWarning(strprintf(_("The following paths are using exFAT, which is known to have intermittent corruption problems on macOS: %s. "
```